CenterForOpenScience/ember-osf-web
# [ENG-4574] Preprint discover rewrite

-   Ticket: [ENG-4574]
-   Feature flag: n/a

## Purpose
- Rewrite the preprints discover page to use the new search-page component

## Summary of Changes
- Add `app/preprints/template.hbs` to house general preprints templating (import CSS using `<ThemStyles>` and `<BrandedNavbar>`
- Update `<BrandedNavbar>` to support preprints in ember-osf-web
- Update preprints discover route and controller to support search-page
- Update search-page component 
  - support default query options to send to SHARE
  - Add result count

## To-Do:
- [x] Invalid preprint provider handling
- [x] Reset theme when leaving route
- [x] Branded header
- [x] Screenshots


## Screenshot(s)
Preprint branded discover page
Desktop
![image](https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/fe4610ba-50d7-41c9-a3b4-dceb0751c7d1)
Mobile
![image](https://github.com/CenterForOpenScience/ember-osf-web/assets/51409893/8aa74a31-b7ad-474a-aa6b-340a0df19971)


## Side Effects
- Should be none...

## QA Notes
- This touches the `<BrandedNavbar>` component that is used in collections. I checked to make sure these didn't break, but a double-check if the buttons for collections navbars are showing up as expected would be much appreciated ("Moderation", "My OSF Projects", "Add to Collection")

[ENG-4574]: https://openscience.atlassian.net/browse/ENG-4574?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
